modules/consensus/accept_reorg_test.go

```diff
@@ -0,0 +1,377 @@
+package consensus
+
+import (
+	"testing"
+
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// reorgSets contains multiple consensus sets that share a genesis block, which
+// can be manipulated to cause full integration blockchain reorgs.
+//
+// cstBackup is a holding place for cstMain - the blocks originally in cstMain get moved
+// to cstBackup so that cstMain can be reorganized without that history being lost.
+// Extending cstBackup will allow cstMain to be reorg'd back to its original blocks.
+type reorgSets struct {
+	cstMain   *consensusSetTester
+	cstAlt    *consensusSetTester
+	cstBackup *consensusSetTester
+}
+
+// createReorgSets creates a reorg set that is ready to be manipulated.
+func createReorgSets(name string) *reorgSets {
+	cstMain, err := createConsensusSetTester(name + " - 1")
+	if err != nil {
+		panic(err)
+	}
+	defer cstMain.closeCst()
+	cstAlt, err := createConsensusSetTester(name + " - 2")
+	if err != nil {
+		panic(err)
+	}
+	defer cstAlt.closeCst()
+	cstBackup, err := createConsensusSetTester(name + " - 3")
+	if err != nil {
+		panic(err)
+	}
+	defer cstBackup.closeCst()
+
+	return &reorgSets{
+		cstMain:   cstMain,
+		cstAlt:    cstAlt,
+		cstBackup: cstBackup,
+	}
+}
+
+// save takes all of the blocks in cstMain and moves them to cstBackup.
+func (rs *reorgSets) save() {
+	for i := types.BlockHeight(1); i <= rs.cstMain.cs.dbBlockHeight(); i++ {
```

Comment:
`dbBlockHeight` executes a read txn every time it's called. Probably best to call it outside the loop, unless the blockheight is always going to be small.